src/java.base/share/classes/com/sun/crypto/provider/CounterMode.java

```diff
@@ -53,6 +53,10 @@ class CounterMode extends FeedbackCipher {
     // number of bytes in encryptedCounter already used up
     private int used;
 
+    // chunkSize is a multiple of block size and used to divide up
+    // input data to trigger the intrinsic.
```

Comment:
For CRC32 digest computation we do support intrinsic at interpreter and c1 compiler level to overcome such warmup related penalties.

Reply:
This is not just a good idea to trigger OSR and therefore use the intrinsic, it's a good idea because very long data causes an extended time to safepoint. I'd support in all cases limiting the size to about a megabyte, which is what we have here.

Reply:
As Andrew points out, giving an intrinsic lots of data, 'backdoors/breaks' a lot of existing algorithms.. from GC not happening because of no safepoint inside the intrinsic, to OSR.. 

.. and (what I believe to be issue for performance here) the call count (CompilationThreshold) to get the intrinsic to compile (well, the callee) in the first place. Though as I pointed in the original issue, I am not entirely convinced it was the call count that got the intrinsic back in; experimentally, chunking got the 'outer intrinsic' to compile. (There is an inner intrinsic that works on 16 byte chunks)

Reply:
Agree with @theRealAph ,  loop induces safe point on back edges which gives opportunity to gc epochs.

Reply:
> As Andrew points out, giving an intrinsic lots of data, 'backdoors/breaks' a lot of existing algorithms.. from GC not happening because of no safepoint inside the intrinsic, to OSR..
> 
> .. and (what I believe to be issue for performance here) the call count (CompilationThreshold) to get the intrinsic to compile (well, the callee) in the first place. Though as I pointed in the original issue, I am not entirely convinced it was the call count that got the intrinsic back in; experimentally, chunking got the 'outer intrinsic' to compile. (There is an inner intrinsic that works on 16 byte chunks)

Please update the comments in the code according.

Reply:
Not sure what about the comment needs to be updated. Maybe provide a suggestion?

Also, please have a look at the original issue, we had a similar discussion about this same comment and this was the result.

Reply:
I dont see any harm in adding descriptive comments giving good justification.

Here is my suggestion:-

"Change facilitate eager intrinsification due to OSR compilation, in addition safe point induced at loop back edge reduce time to saf point before GC epoch."